src/Estimators/StructureFactorEstimator.cpp

```diff
@@ -84,7 +91,20 @@ void StructureFactorEstimator::accumulate(const RefVector<MCPWalker>& walkers,
       sfk_e_e_[k] += weight * (rhok_tot_r_[k] * rhok_tot_r_[k] + rhok_tot_i_[k] * rhok_tot_i_[k]);
       rhok_e_[k] += weight * std::complex<Real>{rhok_tot_r_[k], rhok_tot_i_[k]};
     }
+
+    walkers_weight_ += weight;
+  }
+}
+
+const ParticleSet& StructureFactorEstimator::getParticleSet(const PSPool& psetpool, const std::string& psname) const
```

Comment:
There is `ParticleSetPool::getParticleSet` although the call pattern is different currently.
Once we refine that https://github.com/QMCPACK/qmcpack/issues/5543.
This function can be removed. So some future work needed.

Reply:
Right now I don't want the Estimators directly handling particle set pools. I don't think read consumers of pools need access to anything but a name object map.